client/utils.py

```diff
@@ -148,6 +149,33 @@ def write_command(fdesc, command, destinations, source=0):
     time.sleep(0.3)
 
 
+def write_command_retry(fdesc, command, destinations, source=0):
+    """
+    Writes a command, retries as long as there is no answer and returns a dictionnary containing
+    a map of each board ID and its answer.
+    """
+    write_command(fdesc, command, destinations, source)
+    reader = CANDatagramReader(fdesc)
+    answers = dict()
+
+    while len(answers) < len(destinations):
```

Comment:
>  I don't want half the robot running the new version and half the robot running the old one.

no, the old firmware would already be erased/invalidated.

I prefer half the robot with at least a version rather than a whole robot with just the bootloader.¨
Though, this behavior should be discussed with the rest of the team.